src/AnalysisServices/AnalysisServices.Dataplane/Models/AsAzureDataplaneCmdletBase.cs

```diff
@@ -0,0 +1,176 @@
+
+        /// <summary>
+        /// A DNS safe host name for the server.
+        /// (region.azureAsEndpoint)
+        /// </summary>
+        /// <example>westus.asazure.windows.net</example>
+        protected string DnsSafeHost;
+
+        /// <summary>
+        /// The name of the server.
+        /// </summary>
+        /// <example>testserver</example>
+        protected string ServerName;
+
+        protected override IAzureContext DefaultContext
+        {
+            get
+            {
+                // Nothing to do with Azure Resource Management context
+                return null;
+            }
+        }
+
+        protected override string DataCollectionWarning
+        {
+            get
+            {
+                return Resources.ARMDataCollectionMessage;
+            }
+        }
+
+        // No data collection for this commandlet
+        protected override void InitializeQosEvent() { }
+
+        public IAzureContext CurrentContext
+        {
+            get
+            {
+                if (_currentContext == null)
+                {
+                    _currentContext = AzureRmProfileProvider.Instance.Profile.DefaultContext;
+                }
+
+                return _currentContext;
+            }
+
+            set { _currentContext = value; }
+        }
+
+        public IAsAzureHttpClient AsAzureDataplaneClient
+        {
+            get
+            {
+                if (_asAzureDataplaneClient == null)
+                {
+                    _asAzureDataplaneClient = CreateAsAzureDataplaneClient(DnsSafeHost, CurrentContext, () => { return new HttpClient(); });
+                }
+
+                return _asAzureDataplaneClient;
+            }
+
+            set { _asAzureDataplaneClient = value; }
+        }
+
+        protected override void BeginProcessing()
+        {
+            base.BeginProcessing();
+
+            if (string.IsNullOrEmpty(Instance))
+            {
+                throw new ArgumentNullException(nameof(Instance));
+            }
+
+            // user must specify the fully qualified server name. For example, asazure://westus2.asazure.windows.net/testserver
+            if (!Uri.TryCreate(Instance, UriKind.Absolute, out var uriResult) || uriResult.Scheme != AsAzureEndpoints.UriSchemeAsAzure)
+            {
+                throw new PSInvalidOperationException(string.Format(Resources.InvalidServerName, Instance));
+            }
+
+            // derive all bits of the url from the input
+            ServerUri = uriResult.AbsoluteUri;
+            DnsSafeHost = uriResult.DnsSafeHost;
+            ServerName = uriResult.PathAndQuery.Trim('/');
+
+            if (_asAzureDataplaneClient == null)
+            {
+                AsAzureDataplaneClient = CreateAsAzureDataplaneClient(DnsSafeHost, CurrentContext, () => { return new HttpClient(); });
+            }
+        }
+
+        protected ClusterResolutionResult ClusterResolve(Uri clusterUri, string serverName)
+        {
```

Comment:
This method is called by an overloaded version in Sync-AzureASInstance, which is called in its ExecuteCmdlet(), on line 106. It seems it is used to get different forms of the URIs for the sync.

This method was originally also called in Export-AzureAsInstanceLog, but was unnecessary and I removed it.

I will move this code back into the sync class, since it is not used elsewhere.